packages/block-library/src/paragraph/edit.js

```diff
@@ -108,6 +123,36 @@ function ParagraphBlock( {
 					</ToolsPanelItem>
 				</InspectorControls>
 			) }
+			{ ! content && (
+				<Popover
+					anchorRef={ paragraphElement }
+					animate={ false }
+					position="top right left"
+					focusOnMount={ false }
+					__unstableForcePosition
+					className="wp-block-paragraph__drop_zone"
+				>
+					<DropZone
+						style={ {
+							// TODO: Ideally we should observe the size of the paragraph block.
+							width: paragraphElement?.offsetWidth,
+							height: paragraphElement?.offsetHeight,
+						} }
+						onFilesDrop={ ( files ) => {
+							if ( files.length === 1 ) {
```

Comment:
I think there needs to be a file type check. Though it might be better to check for relevant `file` transforms and trigger the valid ones, so then gallery, video and file blocks will also be supported (as well as any third party blocks that register a valid transform).